src/assets.psm1

```diff
@@ -2126,3 +2126,807 @@ function Remove-SafeguardAssetAccount
 }
+
+<#
+.SYNOPSIS
+Set account SSH Key inside Safeguard for assets under management via the Web API.
+
+.DESCRIPTION
+Set the SSH Key in Safeguard for an account on an asset under management.  This
+just modifies what is stored in Safeguard.  It does not change the actual SSH Key
+of the account.
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER AssetPartition
+An integer containing an ID or a string containing the name of the asset partition
+to set the asset account SSH Key in.
+
+.PARAMETER AssetPartitionId
+An integer containing the asset partition ID to set the asset account SSH Key in.
+(If specified, this will override the AssetPartition parameter)
+
+.PARAMETER AssetToSet
+An integer containing the ID of the asset to set account SSH Key on or a string containing the name.
+
+.PARAMETER AccountToSet
+An integer containing the ID of the account to set SSH Key on or a string containing the name.
+
+.PARAMETER Passphrase
+A SecureString containing the passphrase used to decrypt the private key.
+
+.PARAMETER PrivateKey
+A SecureString containing the SSH Key to assign to the account.
+
+.INPUTS
+None.
+
+.OUTPUTS
+JSON response from Safeguard Web API.
+
+.EXAMPLE
+Set-SafeguardAssetAccountSshKey -AccessToken $token -Appliance 10.5.32.54 -Insecure windows.blah.corp administrator
+
+.EXAMPLE
+Set-SafeguardAssetAccountSshKey -AccountToSet oracle -NewSshKey $sshkey
+#>
+function Set-SafeguardAssetAccountSshKey
+{
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$true)]
+        [object]$AssetPartition,
+        [Parameter(Mandatory=$false)]
+        [int]$AssetPartitionId = $null,
+        [Parameter(Mandatory=$true)]
+        [object]$AssetToSet,
+        [Parameter(Mandatory=$true)]
+        [object]$AccountToSet,
+        [Parameter(Mandatory=$false)]
+        [string]$Passphrase,
+        [Parameter(Mandatory=$true)]
+        [string]$PrivateKey
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    $local:AccountId = (Resolve-SafeguardAssetAccountId -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure `
+                           -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId -Asset $AssetToSet -Account $AccountToSet)
+
+    $local:Body = @{
+        "Passphrase" = $local:Passphrase
+        "PrivateKey" = $local:PrivateKey
+    }
```

Comment:
I guess this works, but is it standard to prefiix cmdlet parameters with local: ? I can see it for $local:Body and $local:AccountId which are being defined here locally but I don't see why it's used for $Passphrase and $PrivateKey that are passed as parameters. This is confusing to me.

Reply:
Fixed, removed local